dEdX/src/dEdxAnalyser.cc

```diff
@@ -531,11 +655,13 @@ void dEdxAnalyser::processEvent( LCEvent * evt ) {
       if (mom > .1) n_track_mom++;
       if (n_track > 1 ) _TrackMomHist->Fill(mom);
       momVec.push_back(mom);
+      int nHits = track->getSubdetectorHitNumbers()[_TPCindex];
 
-      for (int j=0; j<_nPart; j++) if (pdg==_PDG[j] && mom>.3 && mom<100) _dEdxVec[j].push_back(std::array<double,5> {{mom,dEdx,track->getdEdxError(),track->getTanLambda(),(double)track->getSubdetectorHitNumbers()[_TPCindex]}});
+      for (int j=0; j<_nPart; j++) if (pdg==_PDG[j] && mom>.3 && mom<100) _dEdxVec[j].push_back(std::array<double,6> {{mom,dEdx,track->getdEdxError(),track->getTanLambda(),(double)track->getSubdetectorHitNumbers()[_TPCindex],mom_t}});
```

Comment:
```suggestion
      for (int j=0; j<_nPart; j++) {
        if (pdg==_PDG[j] && mom>.3 && mom<100) {
          _dEdxVec[j].emplace_back({mom,dEdx,track->getdEdxError(),track->getTanLambda(),(double)track->getSubdetectorHitNumbers()[_TPCindex],mom_t});
          }
        }
```

Can this be split into several lines? Otherwise this is very error prone and pretty hard to read.

The `emplace_back` should make this a bit more efficient, by constructing the `std::array` in-place instead of constructing and copying afterwards.